libs/wingc/src/jsify.rs

```diff
@@ -411,15 +420,50 @@ impl<'a> JSifier<'a> {
 						.join("\n")
 				)
 			}
-			ExprKind::MapLiteral { fields, .. } => {
-				format!(
-					"Object.freeze(new Map([{}]))",
-					fields
-						.iter()
-						.map(|(key, expr)| format!("[ \"{}\", {} ]", key, self.jsify_expression(expr, phase)))
-						.collect::<Vec<String>>()
-						.join(", ")
-				)
+			ExprKind::JsonLiteral { type_, element } => {
+				*self.is_json.borrow_mut() = true;
+				let json_string = match &element.kind {
+					ExprKind::MapLiteral { type_, fields } => {
+						if is_mutable_collection(expression) {
+							format!("{}", self.jsify_expression(element, phase))
+						} else {
+							format!("Object.freeze({})", self.jsify_expression(element, phase))
+						}
+					}
+					_ => format!("{}", self.jsify_expression(element, phase)),
+				};
+				*self.is_json.borrow_mut() = false;
+				json_string
+			}
+			ExprKind::MapLiteral { fields, type_ } => {
+				if *self.is_json.borrow() {
```

Comment:
Did I understand correctly that we are handling the jsification of json maps in this clause?
I'm a little confused if we should consider `Json` and `MutJson` in here too (then we'll need to check that and apply `Object.freeze()`).

Reply:
I believe since we only need to freeze the whole map or none of it the JsonLiteral ExprKind handles adding `Object.freeze()` then the sub maps dont need to be individually frozen. 

const x = Object.freeze({some: { deep: { object: {}}}});
or
const x = {some: { deep: { object: {}}}};